File: src/ClaudeToolbar.App/Services/ChimePlayer.cs
using System.IO;
using System.Media;
using ClaudeToolbar.Core.Sessions;

namespace ClaudeToolbar.App.Services;

/// <summary>Writes the generated chimes to %LOCALAPPDATA%\ClaudeToolbar\sounds and plays them.</summary>
public sealed class ChimePlayer : IDisposable
{
    private readonly Dictionary<ChimeKind, SoundPlayer> _players = new();

    public ChimePlayer()
    {
        var dir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "ClaudeToolbar", "sounds");
        Directory.CreateDirectory(dir);
        foreach (var kind in Enum.GetValues<ChimeKind>())
        {
            var path = Path.Combine(dir, kind.ToString().ToLowerInvariant() + ".wav");
            File.WriteAllBytes(path, ChimeSynth.Wav(kind));
            var player = new SoundPlayer(path);
            player.Load();
            _players[kind] = player;
        }
    }

    public void Play(ChimeKind kind)
    {
        try
        {
            _players[kind].Play();
        }
        catch (Exception ex)
        {
            Log.Error("Chime failed", ex);
        }
    }

    public void Dispose()
    {
        foreach (var player in _players.Values) player.Dispose();
        _players.Clear();
    }
}

File: mac/Sources/ClaudeToolbar/Services/ChimePlayer.swift
import AppKit
import ClaudeToolbarCore

/// Writes the generated chimes to ~/Library/Application Support/ClaudeToolbar/sounds and plays them.
@MainActor
final class ChimePlayer {
    private var sounds: [ChimeKind: NSSound] = [:]

    init() {
        let base = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first
            ?? FileManager.default.homeDirectoryForCurrentUser
        let directory = base.appendingPathComponent("ClaudeToolbar/sounds", isDirectory: true)
        do {
            try FileManager.default.createDirectory(at: directory, withIntermediateDirectories: true)
        } catch {
            Log.error("Could not create \(directory.path): \(error.localizedDescription)")
            return
        }
        for kind in [ChimeKind.attention, .finished, .failed] {
            let url = directory.appendingPathComponent("\(kind).wav")
            do {
                try ChimeSynth.wav(kind).write(to: url, options: .atomic)
                if let sound = NSSound(contentsOf: url, byReference: false) {
                    sounds[kind] = sound
                } else {
                    Log.error("Could not load chime \(kind) from \(url.path)")
                }
            } catch {
                Log.error("Could not write chime \(kind): \(error.localizedDescription)")
            }
        }
    }

    func play(_ kind: ChimeKind) {
        guard let sound = sounds[kind] else { return }
        sound.stop()
        sound.play()
    }
}
